From da0d25395e12b91a53340943e818a824f5545937 Mon Sep 17 00:00:00 2001
From: Canh Lam <canh.lam.uj@renesas.com>
Date: Fri, 12 Apr 2024 08:16:39 +0700
Subject: [PATCH 1/2] meta-rcar-gateway: Add Docker and kubernetes
 configuration

Add docker.cfg and kubernetes.cfg to serve OCI Container Engine and Container Orchestration

Signed-off-by: Canh Lam <canh.lam.uj@renesas.com>
---
 .../linux/linux-renesas/docker.cfg            | 12 ++++++
 .../linux/linux-renesas/kubernetes.cfg        | 39 +++++++++++++++++++
 .../linux/linux-renesas_%.bbappend            |  7 ++++
 3 files changed, 58 insertions(+)
 create mode 100644 meta-rcar-gateway/recipes-kernel/linux/linux-renesas/docker.cfg
 create mode 100644 meta-rcar-gateway/recipes-kernel/linux/linux-renesas/kubernetes.cfg
 create mode 100644 meta-rcar-gateway/recipes-kernel/linux/linux-renesas_%.bbappend

diff --git a/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/docker.cfg b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/docker.cfg
new file mode 100644
index 00000000..17d59871
--- /dev/null
+++ b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/docker.cfg
@@ -0,0 +1,17 @@
+CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=m
+CONFIG_IP_NF_FILTER=m
+CONFIG_NF_NAT=m
+CONFIG_NF_CONNTRACK=y
+
+CONFIG_DM_THIN_PROVISIONING=m+
+
+CONFIG_IP_NF_NAT=m
+CONFIG_IP_NF_TARGET_MASQUERADE=m
+CONFIG_OVERLAY_FS=y
+CONFIG_VETH=y
+CONFIG_BRIDGE_NETFILTER=y
+CONFIG_NETFILTER_FAMILY_BRIDGE=y
+CONFIG_NETFILTER_XT_TARGET_MASQUERADE=m
+CONFIG_NF_NAT_MASQUERADE=y
+CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=m
+CONFIG_NETFILTER_XT_MATCH_CONNTRACK=m
diff --git a/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/kubernetes.cfg b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/kubernetes.cfg
new file mode 100644
index 00000000..84fa8c57
--- /dev/null
+++ b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas/kubernetes.cfg
@@ -0,0 +1,42 @@
+CONFIG_BLK_CGROUP=y
+CONFIG_BLK_DEV_THROTTLING=y
+CONFIG_CGROUP_PERF=y
+CONFIG_CGROUP_HUGETLB=y
+CONFIG_NET_CLS_CGROUP=y
+CONFIG_CFS_BANDWIDTH=y
+CONFIG_FAIR_GROUP_SCHED=y
+CONFIG_RT_GROUP_SCHED=y
+CONFIG_IP_NF_TARGET_REDIRECT=y
+CONFIG_IP_SET=m
+CONFIG_IP_VS=y
+CONFIG_IP_VS_NFCT=y
+CONFIG_IP_VS_PROTO_TCP=y
+CONFIG_IP_VS_PROTO_UDP=y
+CONFIG_IP_VS_RR=m
+CONFIG_NETFILTER_NETLINK_LOG=m
+CONFIG_NETFILTER_XT_CONNMARK=m
+CONFIG_NETFILTER_XT_MATCH_COMMENT=m
+CONFIG_NETFILTER_XT_MATCH_LIMIT=m
+CONFIG_NETFILTER_XT_MATCH_MARK=m
+CONFIG_NETFILTER_XT_MATCH_PHYSDEV=m
+CONFIG_NETFILTER_XT_TARGET_NFLOG=m
+CONFIG_NETFILTER_XT_TARGET_REDIRECT=m
+CONFIG_NAMESPACES=y
+CONFIG_NET_NS=y
+CONFIG_PID_NS=y
+CONFIG_IPC_NS=y
+CONFIG_UTS_NS=y
+CONFIG_CGROUPS=y
+CONFIG_CGROUP_CPUACCT=y
+CONFIG_CGROUP_DEVICE=y
+CONFIG_CGROUP_FREEZER=y
+CONFIG_CGROUP_SCHED=y
+CONFIG_SMP=y
+CONFIG_CPUSETS=y
+CONFIG_MEMCG=y
+CONFIG_INET=y
+CONFIG_EXT4_FS=y
+CONFIG_PROC_FS=y
+CONFIG_BRIDGE_NETFILTER=y
+CONFIG_NETFILTER_XT_MATCH_MULTIPORT=y
+CONFIG_VXLAN=y
diff --git a/meta-rcar-gateway/recipes-kernel/linux/linux-renesas_%.bbappend b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas_%.bbappend
new file mode 100644
index 00000000..c3cb3144
--- /dev/null
+++ b/meta-rcar-gateway/recipes-kernel/linux/linux-renesas_%.bbappend
@@ -0,0 +1,7 @@
+SRC_URI += " \
+    file://docker.cfg \
+    file://kubernetes.cfg \
+"
+
+K3S_CONFIG_FILE_MD5 = "e9e92ca071e15ccf376c85369985076d"
+CONTAINERS_CONFIG_FILE_MD5 = "cb94a5c5c9e4260ce2045dd3e3548656"
-- 
2.25.1

